src/Project.js

```diff
@@ -633,6 +637,19 @@ class Project extends DirItem {
         } else {
             exitValue = this.resultStats.errors > 0 ? 2 : ((this.resultStats.warnings > 0) ? 1 : 0);
         }
+
+        if (this.options.opt.output) {
+            if (typeof this.formatter.writeSummaryToFile === 'function') {
+                let summary = this.formatter.writeSummaryToFile(this.project.name, totalTime, this.fileStats, this.resultStats, score) ;
```

Comment:
If you are going to call a function in a formatter, the abstract version of this function should be defined in the Formatter superclass in ilib-lint-common along with the definitions and types of all of the parameters so that any formatter (not just the webos one) can perform this function.

Also, this function does not write to a file, so the name is not accurate. Instead, it should be called something like "formatSummary" to go along with the "format" method for results.

Also, in order to minimize the changes we need to do to the ilib-lint-common in the future, the parameters to this function should be passed in an object instead of individual parameters:

```javascript
let summary = this.formatter.formatSummary({
  name: this.project.name,
  totalTime,
  fileStats: this.fileStats
  resultStats: this.resultStats,
  score
});
```

Reply:
I updated it to pass the options through the object.